spec/support/rubocop_spec_helper.rb

```diff
@@ -4,7 +4,7 @@ def rubocop_results
     rubocop = JSON.parse(`rubocop --format=json --no-display-cop-names #{rubocop_check_path}`)
     hamllint = JSON.parse(`haml-lint --reporter=json #{rubocop_check_path}`)
```

Comment:
So we are probably going to need to change these two lines in the future to have the run in a different form.  I manually fixed the files to get them into the "Pronto format", but that is not a good long term solution.

A few things worth noting here though:

- The `spec/workers/commit_monitor_handlers/commit_range/rubocop_checker/data/*/` directories are not git dirs, so they don't play well with Pronto's runners, since they assume `git`.
- I had two potential approaches two this:
  - Convert those directories to git, and use a `FakeResults` class (or something) that inherits from the `CodeAnalysisMixin` to share that code
  - Create a set of fake libs for pronto that stub the git portions of the code to get the results we need.

I am not sure what is better at this point, because while the `git` approach is probably less code overall, we would either have to do a `git init && git add . && git commit -m "Initial"` as part of the block, or commit the `.git` dir for each one of these directories, which is... fun.  The other option adds a bunch of code we need to support, so not really a big fan of that either.

But decided to sideline fixing that for now to move this forward.

Reply:
> Convert those directories to git, and use a FakeResults class (or something) that inherits from the CodeAnalysisMixin to share that code

This sounds good to me, though I don't know why we need the FakeResults class?

Reply:
> This sounds good to me, though I don't know why we need the FakeResults class?

Because `CodeAnalysisMixin` is a module, and we need to be able to set `@branch` and such to be able to use `.pronto_result`

https://github.com/ManageIQ/miq_bot/blob/6bd031c9154554b73824ae9561fac1dfbfaef5dc/app/workers/concerns/code_analysis_mixin.rb#L32-L39

> We do this in ManageIQ proper, and we didn't have an issue with it, I think

Where?  I am curious.

Also, not saying it isn't feasible, but it is kinda ugly checking in a bunch of extra git objects just to view a diff.  I think I would rather have the `unless File.exists?` be a bit smarter, and do the git repo initialization inside the block (if regenerating, and blow it away if it exists), but not check it in to this repo.

Reply:
> > We do this in ManageIQ proper, and we didn't have an issue with it, I think
> 
> Where?  I am curious

Okay, I think you are talking about these:

https://github.com/ManageIQ/manageiq/tree/master/spec/fixtures/git_repos

Which those are bare repos.  Probably would still work, but would rather avoid checking in a bunch of nest git files, as show for on of those:

```
$ tree spec/fixtures/git_repos/no_master.git/
spec/fixtures/git_repos/no_master.git/
├── HEAD
├── config
├── objects
│   ├── 43
│   │   └── 0ff1dcee43d2e8a12c3ba36f5ff20e46136152
│   ├── 44
│   │   └── f84e202a8eb9d8d35fc183ae8637db3b003aa8
│   ├── 78
│   │   └── 3878cb3ed5964c75767141176c80f7d528abb5
│   ├── 95
│   │   └── 99fdeb034597d90d72d2f58396dee096885b79
│   ├── ae
│   │   └── fde3a01f6e10d72fd4899ce14c8b2654d3eb45
│   ├── e3
│   │   └── 16f7d7548ea8f874c872ff3be04f4e9849833b
│   ├── e6
│   │   └── 9de29bb2d1d6434b8b29ae775ad8c2e48c5391
│   └── fd
│       └── b4bc8c22f4b3ca8279c2e1eb92670893473d05
├── packed-refs
└── refs
    └── tags
        ├── tag1
        └── tag2

11 directories, 13 files
```

Just for a few specs, especially when the `git` part is very inconsequential.  It also seems like it would be hard to modify as end users as well.

Reply:
Yeah those are it.  FWIW, I think having those in core is also weird, so maybe whatever you choose we can do something similar in core as well.